packages/fixie-sdk/src/fixie-serve-bin.tsx

```diff
@@ -79,10 +79,15 @@ async function serve({
           Readable.from(
             (async function* () {
               let lastMessages = [];
+              let currentValue = undefined as string | undefined;
               while (true) {
-                let currentValue = undefined as string | undefined;
                 try {
                   const next = await generator.next();
+                  if (!next.done && currentValue == next.value) {
+                    // No change, don't send anything.
+                    continue;
+                  }
```

Comment:
Q: Is this just for performance optimization or is it needed for correctness too?

Reply:
Good question, it's just an optimization.